File: crates/codeilus-parse/tests/fixtures/sample.ts
import { readFile } from 'fs/promises';

interface Reader {
  read(): Promise<string>;
}

class FileReader implements Reader {
  constructor(private path: string) {}

  async read(): Promise<string> {
    return readFile(this.path, 'utf-8');
  }
}

export function process(reader: Reader): Promise<string> {
  return reader.read();
}

File: crates/codeilus-parse/tests/fixtures/sample.rs
use std::fs;

pub struct Config {
    pub path: String,
}

impl Config {
    pub fn load(path: &str) -> Self {
        Self { path: path.to_string() }
    }

    pub fn read(&self) -> String {
        fs::read_to_string(&self.path).unwrap_or_default()
    }
}

pub fn process(config: &Config) -> String {
    config.read().to_uppercase()
}
